vayesta/core/types/wf/ccsdtq.py

```diff
@@ -35,5 +35,24 @@ def as_ccsd(self):
     def as_cisd(self, c0=1.0):
         return self.as_ccsd().as_cisd()
 
-class UCCSDTQ_WaveFunction(RCCSDTQ_WaveFunction):
-    pass
+class UCCSDTQ_WaveFunction(wf_types.WaveFunction):
+    def __init__(self, mo, t1, t2, t3, t4):
+        super().__init__(mo)
+        self.t1 = t1
+        self.t2 = t2
+        self.t3 = t3
+        self.t4 = t4
+        # TODO
+        #if not (isinstance(t4, tuple) and len(t4) == 2):
+        #    raise ValueError("t4 definition in RCCSDTQ wfn requires tuple of (abaa, abab) spin signatures")
+
+    def as_ccsdtq(self):
+        return self
+
+    def as_ccsd(self):
+        if self.projector is not None:
+            raise NotImplementedError
+        return wf_types.UCCSD_WaveFunction(self.mo, self.t1, self.t2)
+
+    def as_cisd(self, c0=1.0):
+        return self.as_ccsd().as_cisd()
```

Comment:
`as_ccsdtq` and `as_cisd` should be put into the RCCSDTQ_WaveFunction class - for `as_ccsd`, do we not have a `CCSD_WaveFunction`  function, which automatically determines R vs U? Then this could also be put into RCCSDTQ